### PR TITLE
Add the logtostdout and colorlogtostdout flag to allow logging to stdout

### LINKED ISSUE #701: Console logging: separation of stdout and stderr
For console logging, it seems like there is no way to configure Glog to send non-error logs to `stdout` and error logs to `stderr`. 

By default, Glog sends only error logs to stderr, and there is a `FLAGS_logtostderr` which sends ALL logs to `stderr` which is not ideal.  It will be good to have a flag to log to _both_ stdout and stderr appropriately.

### PR DESCRIPTION
**Pull request by git-hulk**
Currently, glog allowed to use logtostderr to send error logs to stderr,
but many log tailors would regard logs from stderr as error logs. So we
want to send non-error logs to stdout and only send error logs to stderr
according to the stderrthreshold.

Resolved #701

### CODE REVIEW

**Comment by sergiud** [comment]
The tests of MSVC builds are failing. Could you please rebase onto current master and see if this helps?

**Comment by git-hulk** [comment]
@sergiud cool, thanks for your great help!

**Comment by git-hulk** [comment]
@sergiud How do you think about this PR? Does this implementation make sense for you?

**Comment by sergiud** [comment]
I already skipped through the PR and it does make sense to me. However, I would like to wait for additional comments from users (if any). So please hold tight.

**Comment by git-hulk** [comment]
ok, cool. Sorry for pushing you.

File: src/googletest.h [hunk 1]
```diff
@@ -536,6 +546,14 @@ static inline bool MungeAndDiffTestStderr(const string& golden_filename) {
   return true;
 }
 
+static inline bool MungeAndDiffTestStderr(const string& golden_filename) {
+  return MungeAndDiffTestStderrOrStdout(golden_filename, false);
```
Comment (sergiud) [inline src/googletest.h:550]: `bool` parameters are hard to follow. It would be clearer if you pass `CapturedStream* cap` or an index into `s_captured_streams` to avoid code repetition.

File: src/logging.cc [hunk 2]
```diff
@@ -739,43 +743,61 @@ inline void LogDestination::SetEmailLogging(LogSeverity min_severity,
   LogDestination::addresses_ = addresses;
 }
 
-static void ColoredWriteToStderr(LogSeverity severity,
-                                 const char* message, size_t len) {
-  const GLogColor color =
-      (LogDestination::terminal_supports_color() && FLAGS_colorlogtostderr) ?
-      SeverityToColor(severity) : COLOR_DEFAULT;
+static void ColoredWriteToStderrOrStdout(bool is_stdout, LogSeverity severity,
```
Comment (sergiud) [inline src/logging.cc:746]: Instead of a `bool` you can pass a `FILE*` directly.

File: src/logging.cc [hunk 3]
```diff
@@ -739,43 +743,61 @@ inline void LogDestination::SetEmailLogging(LogSeverity min_severity,
...
+                                         const char* message, size_t len) {
+  // We also need to send logs to the stderr when the severity is
+  // higher or equal to the stderr threshold.
+  is_stdout = is_stdout && severity < FLAGS_stderrthreshold;
```
Comment (sergiud) [inline src/logging.cc:750]: I think the decision whether the output is written `stdout` or `stderr` should be made outside of this function. You can introduce a helper function if you need to.

File: src/logging.cc [hunk 4]
```diff
@@ -739,43 +743,61 @@ inline void LogDestination::SetEmailLogging(LogSeverity min_severity,
...
   // exit code, and cerr may be partially or fully destroyed by then.
   if (COLOR_DEFAULT == color) {
-    fwrite(message, len, 1, stderr);
+    fwrite(message, len, 1, output_fd);
     return;
   }
 #ifdef GLOG_OS_WINDOWS
-  const HANDLE stderr_handle = GetStdHandle(STD_ERROR_HANDLE);
+  const HANDLE output_handle =
+      GetStdHandle(is_stdout ? STD_OUTPUT_HANDLE : STD_ERROR_HANDLE);
```
Comment (sergiud) [inline src/logging.cc:767]: Here you can distinguish between `stderr` and `stdout` without relying on a `bool`.

**Review by sergiud — CHANGES_REQUESTED** [review]
I have a few suggestions. Thanks for looking into the issue!

**Comment by sergiud** [comment]
Looks great!

**Review by sergiud — APPROVED** [review]
